/*
 * ARX: Powerful Data Anonymization
 * Copyright 2012 - 2015 Florian Kohlmayer, Fabian Prasser
 * 
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.deidentifier.arx.test;

import java.io.IOException;

import org.deidentifier.arx.examples.Example1;
import org.deidentifier.arx.examples.Example10;
import org.deidentifier.arx.examples.Example11;
import org.deidentifier.arx.examples.Example12;
import org.deidentifier.arx.examples.Example13;
import org.deidentifier.arx.examples.Example16;
import org.deidentifier.arx.examples.Example17;
import org.deidentifier.arx.examples.Example18;
import org.deidentifier.arx.examples.Example19;
import org.deidentifier.arx.examples.Example2;
import org.deidentifier.arx.examples.Example20;
import org.deidentifier.arx.examples.Example21;
import org.deidentifier.arx.examples.Example22;
import org.deidentifier.arx.examples.Example23;
import org.deidentifier.arx.examples.Example24;
import org.deidentifier.arx.examples.Example25;
import org.deidentifier.arx.examples.Example26;
import org.deidentifier.arx.examples.Example3;
import org.deidentifier.arx.examples.Example4;
import org.deidentifier.arx.examples.Example5;
import org.deidentifier.arx.examples.Example6;
import org.deidentifier.arx.examples.Example7;
import org.deidentifier.arx.examples.Example8;
import org.deidentifier.arx.examples.Example9;
import org.junit.Assert;
import org.junit.Test;

/**
 * 
 */
public class TestExamples {

    /**
     * 
     */
    @Test
    public void testExample1() {
        try {
            Example1.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    /**
     * 
     *
     * @throws IOException
     */
    @Test
    public void testExample2() throws IOException {
        try {
            Example2.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    /**
     * 
     */
    @Test
    public void testExample3() {
        try {
            Example3.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    /**
     * 
     */
    @Test
    public void testExample4() {
        try {
            Example4.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    /**
     * 
     */
    @Test
    public void testExample5() {
        try {
            Example5.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    /**
     * 
     */
    @Test
    public void testExample6() {
        try {
            Example6.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    /**
     * 
     */
    @Test
    public void testExample7() {
        try {
            Example7.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    /**
     * 
     */
    @Test
    public void testExample8() {
        try {
            Example8.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    /**
     * 
     */
    @Test
    public void testExample9() {
        try {
            Example9.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    /**
     * 
     */
    @Test
    public void testExample10() {
        try {
            Example10.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    /**
     * 
     */
    @Test
    public void testExample11() {
        try {
            Example11.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    /**
     * 
     */
    @Test
    public void testExample12() {
        try {
            Example12.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    /**
     * 
     */
    @Test
    public void testExample13() {
        try {
            Example13.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    /**
     * 
     */
    @Test
    public void testExample16() {
        try {
            Example16.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    /**
     * 
     */
    @Test
    public void testExample17() {
        try {
            Example17.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    /**
     * 
     */
    @Test
    public void testExample18() {
        try {
            Example18.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    /**
     * 
     */
    @Test
    public void testExample19() {
        try {
            Example19.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    /**
     * 
     */
    @Test
    public void testExample20() {
        try {
            Example20.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    /**
     * 
     */
    @Test
    public void testExample21() {
        try {
            Example21.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }
    
    /**
     * 
     */
    @Test
    public void testExample22() {
        try {
            Example22.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }
    
    /**
     * 
     */
    @Test
    public void testExample23() {
        try {
            Example23.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }
    
    /**
     * 
     */
    @Test
    public void testExample24() {
        try {
            Example24.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }
    
    /**
     * 
     */
    @Test
    public void testExample25() {
        try {
            Example25.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }
    
    /**
     * 
     */
    @Test
    public void testExample26() {
        try {
            Example26.main(null);
        } catch (final Exception e) {
            e.printStackTrace();
            Assert.fail();
        }
    }
}
